src/9_besoin_ch.js

```diff
@@ -101,9 +99,14 @@ export default function calc_besoin_ch(
     // pertes generation
     const Bch_hp_j = bvj * dh19j;
     const Bch_hp_j_dep = bvj_dep * dh21j;
-    pertes_generateur_ch_recup += calc_Qrec_gen_j(gen_ch, nref19, Bch_hp_j) / (1000 * 1000);
-    pertes_generateur_ch_recup_depensier +=
-      calc_Qrec_gen_j(gen_ch, nref21, Bch_hp_j_dep) / (1000 * 1000);
+
+    for (const inst_ch of instal_ch) {
```

Comment:
Il faudrait récupérer la liste des générateurs avant la boucle des mois pour éviter ces "boucles for" 12 fois

Reply:
Je crois meme qu'il doit y a avoir des générateurs à exclure car il n'y a pas de recup dessus, mais je ne trouve pas la doc là dessus. Tu re réfère à ce doc ? https://rt-re-batiment.developpement-durable.gouv.fr/IMG/pdf/consolide_annexe_1_arrete_du_31_03_2021_relatif_aux_methodes_et_procedures_applicables.pdf il date d'oct 2021 y'a pas eu de mise à jour depuis ?

Reply:
Oui les générateurs à exclure sont ceux qui ne sont pas en volume chauffé position_volume_chauffe === 0 dans la méthode calc_Qrec_gen_j (cette variable est par contre utilisée dans certains DPE à 0, dans d'autres à 1 pour la même chose, j'ai commencé à mettre des méthodes de récupérations des données dans les sous fiches techniques pour valider cet info, les données des fiches techniques étant plus explicites, il est possible d'avoir une info plus précise sur l'utilisation des variables comme celle-ci). Par exemple pour la présence d'un comptage ou alors pour plusieurs_facade_exposee les valeurs sont confrontées aux fiches techniques. (le boolean bug_for_bug_compat permet dans un premier temps de bien conditionner ces cas de figure)

Oui je prends cette doc 
https://rt-re-batiment.developpement-durable.gouv.fr/IMG/pdf/consolide_annexe_1_arrete_du_31_03_2021_relatif_aux_methodes_et_procedures_applicables.pdf

Il n'y a eu que certains amendements notamment sur les seuils des lettres de sorties pour les surfaces < 40m2

Reply:
![image](https://github.com/user-attachments/assets/5354a596-8588-4538-98cf-7101b91615a1)

Reply:
Done